src/include/ipc4/detect_test.h

```diff
@@ -0,0 +1,56 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2022 Intel Corporation. All rights reserved.
+ *
+ * Author: Ievgen Ganakov <ievgen.ganakov@intel.com>
+ */
+
+/**
+ * \file include/ipc4/detect_test.h
+ * \brief KD module definitions.
+ * NOTE: This ABI uses bit fields and is non portable.
+ */
+
+#ifndef __SOF_IPC4_DETECT_TEST_H__
+#define __SOF_IPC4_DETECT_TEST_H__
+
+#include "base-config.h"
+
+struct ipc4_detect_test_module_cfg {
```

Comment:
Why do we need two structures that share the same fields? all the fields defines in this structure are all the same with `struct sof_detect_test_config`,

Reply:
The main reason was to avoid additional `#ifdef`'s while handling KD params. I can remove `struct ipc4_detect_test_module_cfg` and cast using `struct sof_detect_test_config`